tls/understand-tls/understand-tls.tex

```diff
@@ -0,0 +1,275 @@
+confirm that they are talking to the server that they think they are (and not a 
+malicious attacker pretending to be that server). Frequently higher level 
+application protocols may add the capability to authenticate clients, although
+it is also possible to do it at the SSL/TLS layer if so desired.
+
+Identity is established through the use of a \emph{digital certificate}. The 
+digital certificate provides public data about a server for which it is issued. 
+For example, the certificate will contain the hostname(s) for which it is 
+valid. In order to obtain a certificate a server operator must first create a 
+private and a public key. The private key must remain secret. Loss of the 
+private key would be catastrophic to the security of the system. Anyone with 
+access to the private key can masquerade as the server. The public key is 
+mathematically related to the private key, although it is not possible to 
+derive the private key from it. The public key is published in the digital 
+certificate and it is safe for this to be available to everyone.
+
+Having obtained a private/public key pair a server operator must obtain their
+certificate from a \emph{Certificate Authority} (CA)\footnote{CAs can be
+privately run and purely internal to an organisation; or they could be public.
+Which type is most appropriate will depend on what the certificte will be used
+for. There are many public CAs available. A simple search in your search engine
+of choice should turn up lots of links. Many, but not all, charge a fee for
+issuing a certificate.}. The CA will, at a minimum, verify that the server 
+operator is in control of the domain name to be included in the certificate. 
+Dependant on the type of certificate ordered, other checks may also be 
+performed to verify the identity of the server operator. Finally the CA will 
+issue the certificate which itself will be digitally signed by the CA.
+
+Both the digital certificate, and the associated private key are installed on 
+the SSL/TLS server. When a client accesses the server, the server will send 
+its certificate back to the client. In order for authentication to be
+successful the client must verify two things:
+\begin{enumerate}
+\item The certificate provided by the server is valid and issued by a CA that 
+the client trusts.
+\item The server has the private key corresponding to the public key published 
+in the certificate.
+\end{enumerate}
+
+Part of the role of the SSL/TLS protocol is to enable the client to perform the 
+above checks during the establishment of a connection. If either of these 
+checks fail then the connection will fail.
+
+As mentioned above it is also possible for the server to authenticate the 
+client as part of the SSL/TLS protocol. If this capability is used then it works
+in a very similar way  to that described above. The primary difference is that
+the client will also have to create a private/public key pair and obtain a
```

Comment:
I guess this could also need to consider the scenario without key generation by users. Same as above discussion.

Reply:
Hmmm...maybe...but this seems less likely to be automated by a cloud provider? If someone is planning on setting up a client auth infrastructure then they're going to need to have a really good understanding on what is really happening.